### ISSUE TITLE
New protocol: Implement mod gathering

### ISSUE THREAD

**Issue by lionkor** [BeamMP/BeamMP-Server#279]
The server needs to collect all mods (.zip files) from the $RESOURCE_FOLDER/Client folder. 

Acceptance criteria
---
- [ ] Send list of mods in a special format to the backend in heartbeats (see existing code in THeartbeat).
- [ ] Each mod entry needs to also store: Hash of the mod, algorithm used to hash the mod (must currently be SHA256), file size, file name (separate from file location), and whether the mod is optional (always false). Also needs to store the method of getting the mod, currently that is "direct" (make an enum).
- [ ] Add it so that support for possibly hot-reloading mods (update hash, size), and adding mods on the fly could be done.
- [ ] Optional: Check that the mod is not empty, warn on empty folders or other files in the folder that are not .zip files.

**Comment by my-name-is-samael** [comment]
bump ! I would love to have mod adding on the fly ❤️

I posted an issue on BeamMP but it is more relevant in here => https://github.com/BeamMP/BeamMP/issues/577

TLDR: I want to add programmatically modded maps to switch map ingame (manually or with vote), but I don't want players to download _EVERY_ maps on their first connection (can be too heavy).
If changing map without reconnecting players would be tough (and I understand it would be), having client mods change on the fly would be awesome

**Comment by my-name-is-samael** [comment]
Oh btw I forgot:
If an inspection of mods archives is possible, maybe detect map ones and not serve those that are not currently in use to preserve bandwidth and avoid players having to download 50+ maps before connecting
Sorry if I am not clear enough

**Comment by lionkor** [comment]
Makes sense, thanks for your input :)

**Comment by IngvarV8** [comment]
Hello, I would like to work on that issue if nobody is assigned

**Comment by lionkor** [comment]
Hi, this is for the new protocol which is on hold at the moment:)
